server/app/services/agent/node_plugger.rb

```diff
@@ -18,6 +18,7 @@ def plugin!
       begin
         prev_seen_at = node.last_seen_at
         self.update_node
+        self.publish_update_event
```

Comment:
So if I understand this correctly, changing the `update_node` method to use `node.update_attributes(...)` would also trigger the `publish_update_event`?